app/src/main/java/com/kpiroom/bubble/source/api/impl/firebase/FirebaseDbUtil.kt

```diff
@@ -57,4 +69,35 @@ class FirebaseDbUtil(val firebaseDb: FirebaseDatabase) {
         }
         ref.addListenerForSingleValueEvent(listener)
     }
+
+    private suspend fun <T : Any?> readIfConnected(
+        path: String,
+        type: Class<T>
+    ): T =
+        if (isConnected())
+            read(path, type)
+        else
+            throw FirebaseNetworkException(str(R.string.db_no_connection))
```

Comment:
Message must be initialized in your custom exception.
Flow:
1. throw FirebaseNetworkException
2. AsyncProcessor: handle and transform FirebaseNetworkException -> YourCusomException(CODE_NO_INTERNET)
3. AsyncProcessor.handleError({ showError(it.message) }) //And here we retrieve message. In the end of whole chain.

Reply:
Exactly like it works